Lib/webbrowser.py

```diff
@@ -15,14 +15,15 @@ class Error(Exception):
 
 _browsers = {}          # Dictionary of available browser controllers
 _tryorder = []          # Preference order of available browsers
+_preferred_browser = None # The preferred browser for the current environment
 
-def register(name, klass, instance=None, update_tryorder=1):
-    """Register a browser connector and, optionally, connection."""
+def register(name, klass, instance=None, priority=False):
+    """Register a browser connector."""
     _browsers[name.lower()] = [klass, instance]
-    if update_tryorder > 0:
-        _tryorder.append(name)
-    elif update_tryorder < 0:
+    if priority or (_preferred_browser and name in _preferred_browser):
```

Comment:
keyword-only is our preferred approach to handling boolean arguments in Python 3, as otherwise you end up with code that's ambiguous at the point of calling (as here, where the `True` conveys no info to a reader that doesn't already know the signature, while `preferred=True` is a lot more descriptive)

Regarding `_preferred_browser` potentially needing to match things like `firefox` against `firefox.desktop`, that belongs in an explanatory comment in the code itself. I'm fine with dropping the idea of switching to a set, though.